idea/src/org/jetbrains/kotlin/idea/quickfix/AddPropertyToSupertypeFix.kt

```diff
@@ -0,0 +1,191 @@
+            val property = diagnostic.psiElement as? KtProperty ?: return null
+
+            val descriptors = generatePropertiesToAdd(property)
+            if (descriptors.isEmpty()) return null
+
+            val project = diagnostic.psiFile.project
+            val propertyData = descriptors.mapNotNull { createPropertyData(it, property.initializer, project) }
+            if (propertyData.isEmpty()) return null
+
+            return AddPropertyToSupertypeFix(property, propertyData)
+        }
+
+        private fun createPropertyData(
+            propertyDescriptor: PropertyDescriptor,
+            initializer: KtExpression?,
+            project: Project
+        ): PropertyData? {
+            val classDescriptor = propertyDescriptor.containingDeclaration as ClassDescriptor
+            var signaturePreview = IdeDescriptorRenderers.SOURCE_CODE_SHORT_NAMES_NO_ANNOTATIONS.render(propertyDescriptor)
+            if (classDescriptor.kind == ClassKind.INTERFACE) {
+                signaturePreview = signaturePreview.substring("abstract ".length)
+            }
+
+            var sourceCode = IdeDescriptorRenderers.SOURCE_CODE.render(propertyDescriptor)
+            if (classDescriptor.kind == ClassKind.CLASS && classDescriptor.modality == Modality.OPEN && initializer != null) {
+                sourceCode += " = ${initializer.text}"
+            }
+
+            val targetClass = DescriptorToSourceUtilsIde.getAnyDeclaration(project, classDescriptor) as? KtClass ?: return null
+            return PropertyData(
+                signaturePreview,
+                sourceCode,
+                targetClass
+            )
+        }
+
+        private fun generatePropertiesToAdd(propertyElement: KtProperty): List<PropertyDescriptor> {
+            val propertyDescriptor = propertyElement.resolveToDescriptorIfAny(BodyResolveMode.FULL) ?: return emptyList()
+            val classDescriptor = propertyDescriptor.containingDeclaration as? ClassDescriptor ?: return emptyList()
+
+            return getSuperClasses(classDescriptor)
+                .asSequence()
+                .filterNot { KotlinBuiltIns.isAnyOrNullableAny(it.defaultType) }
+                .map { generatePropertySignatureForType(propertyDescriptor, it) }
+                .toList()
+        }
+
+        private fun getSuperClasses(classDescriptor: ClassDescriptor): List<ClassDescriptor> {
+            val supertypes = classDescriptor.defaultType.supertypes().toMutableList().sortSubtypesFirst()
+            return supertypes.mapNotNull { it.constructor.declarationDescriptor as? ClassDescriptor }
+        }
+
+        private fun MutableList<KotlinType>.sortSubtypesFirst(): List<KotlinType> {
+            val typeChecker = KotlinTypeChecker.DEFAULT
+            for (i in 1 until size) {
+                val currentType = this[i]
+                for (j in 0 until i) {
+                    if (typeChecker.isSubtypeOf(currentType, this[j])) {
+                        this.removeAt(i)
```

Comment:
Your manual sorting method has estimated performance of `O(N^3)` because of this `removeAt`, and this could slow everything down.